### PR TITLE
[fireurld.service] remove old socket

### PR DESCRIPTION
**Pull request by 4parthy**
Solves #15

### CODE REVIEW

File: systemd/fireurld.service [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Description=fireurl deamon
 
 [Service]
+ExecStartPre=/usr/bin/rm -f ${XDG_RUNTIME_DIR}/fireurl/fireurl0
```
Comment (rusty-snake) [inline systemd/fireurld.service:5]:
```suggestion
ExecStartPre=-/usr/bin/rm -f ${XDG_RUNTIME_DIR}/fireurl/fireurl0
```

**Review by rusty-snake — COMMENTED** [review]
> Solves #15 

That's the question, does it solve it or is it just a workaround. i.e. do we want to merge this as a fix or do we want a different fix.

The other options we have are
1. do it at `ExecStopPost=` (together with `Restart=on-failure`).
2. catch signals and delete before exiting.

ping @a-ilin @Yarn for opinions.

- `ExecStartPre` could take over a path that is actually used by a fireurld that was started by a different mechanism than systemd, say autostart.
- `ExecStartPre`/`ExecStopPost` will be hard to adopt when we add a `FIREURLD_SOCKET_PATH` environment variable unless we set it in the service unit.
- `ExecStartPre`/`ExecStopPost` do not solve the issue for fireurld started by non-systemd mechanisms such as autostart.
- catching signals solves all of the above but does not work for `SIGKILL`.

**Comment by 4parthy** [comment]
It is a workaround and signals seems the right way to handle different init systems / autostart imho. No idea about how to handle cleanup after a SIGKILL right now but I believe there are some possible solution.
To be honest I opened this pull request because I adopted the arch package which shipped its own service file with this workaround and I'd rather have it maintained here by the community. 
Maybe a compromise could be to push this workaround for now and keep the Issue open to work on a better solution.

**Comment by rusty-snake** [comment]
> arch package 

wait, there's an Arch package for it?

... https://aur.archlinux.org/packages/fireurl

need to update the readme

> Maybe a compromise could be to push this workaround for now and keep the Issue open to work on a better solution.

Yes. Still we can evaluate Pre or Post cleanup (+ Restart if we want).

I'll wait some time to give other chance to comment and merge it then.

**Comment by 4parthy** [comment]
> wait, there's an Arch package for it?
> 
> ... https://aur.archlinux.org/packages/fireurl

sorry, aur package, didn't mean to spook :x

> Yes. Still we can evaluate Pre or Post cleanup (+ Restart if we want).
> 
> I'll wait some time to give other chance to comment and merge it then.

agreed

**Comment by rusty-snake** [comment]
> sorry, aur package, didn't mean to spook :x

No I meant Arch/AUR I.e. packaged somewhere even launchpad or cope would count.

**Comment by Yarn** [comment]
> ExecStartPre could take over a path that is actually used by a fireurld that was started by a different mechanism than systemd, say autostart.

This is true for manually deleting it as well, which seems to be the current option.

Actual graceful shutdown on a signal seems like the correct solution long term but this seems like the most reasonable stopgap.

**Pull request merged**